expr/functions.go

```diff
@@ -15,9 +15,93 @@ var ErrWrongArgc = errors.New("wrong number of arguments")
 var ErrBadArgument = errors.New("bad argument")
 
 var allFns = map[string]Function{
+	"Math.max":  mathMax,
+	"Math.min":  mathMin,
 	"Math.sqrt": mathSqrt,
 }
 
+func mathMax(args []zngnative.NativeValue) (zngnative.NativeValue, error) {
+	if len(args) == 0 {
```

Comment:
Could we encode arity elsewhere (for example in `allFns`) and check this at compile time?

Reply:
Yes, I had actually done that in a draft of the first functions PR and then decided to switch it this way so that all the argument validation (count, types, etc.) remains in one place.  I'm happy to revisit it but at the moment, don't see a huge advantage for either approach.

Reply:
> don't see a huge advantage for either approach

I might have been making an incorrect assumption here: with the way things are currently done, doesn't `fmt.Errorf("Math.max: %w", ErrWrongArgc)` end up as a runtime warning ? If that's the case, then it seems to me like this should fail at compile time since the error is a property of the program regardless of the data. If that's not the case, then I agree there's no major advantage to either approach...

Reply:
Oh yes, good point.  I think we still want the runtime warnings to be useful and prominent here but in the short term, I think you're right.  Will update.

Reply:
👍 

> we still want the runtime warnings to be useful and prominent

Out of curiosity, is this in anticipation of variadic args (e.g `Math.max(some_array...)`)?  That was the one situation I could think of where compile-time wouldn't suffice...

Reply:
Nothing so fancy.  Just that if you mistype a field name or apply a function to a value of the wrong type, that can't be detected until runtime and it would be a pain to debug such a query without warnings to point toward the problem.

Reply:
Ah I see, agree that those should continue to be runtime warnings (I was thinking specifically about arity-related issues).